mautic/mautic
# Fix migrations preup with table prefix and migration performance

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [Y]
| New feature/enhancement? (use the a.x branch)      | [N]
| Deprecations?                          | [N]
| BC breaks? (use the c.x branch)        | [N]
| Automated tests included?              | [N] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:

This PR fixes issue with correct preUp migration conditions with and without prefix.
Also optimise query generation

#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Run the migrations on old version of Mautic and check migrations are successful




<a href="https://gitpod.io/#https://github.com/mautic/mautic/pull/11371"><img src="https://gitpod.io/button/open-in-gitpod.svg"/></a>
